compiler/hash-tir/src/pats.rs

```diff
@@ -40,10 +40,10 @@ pub struct Spread {
 #[derive(Copy, Clone, Debug)]
 pub struct RangePat {
     /// The beginning of the range.
-    pub lo: LitPat,
+    pub lo: Option<LitPat>,
```

Comment:
Why is this an option, why not inline min/max during resolution?

Reply:
Yeah I thought this would be the best thing to do, it would avoid quite alot of code duplication. I just wasn’t sure if it made sense to do it in resolution. Ideally it should happen when the type of each operand is known. This would avoid the changes in lowering and exhaustiveness. What do you think?

Reply:
Since the type should be known it should happen during inference as you said. However is there any time the type will not be known because it will be a generic?

Reply:
Well, for now it is only possible for the range to be composed of numeric literals, hence inly `char` and sized `{i|u}xxx` types are supported.